packages/core/src/modules/mdoc/MdocService.ts

```diff
@@ -0,0 +1,78 @@
+import type {
+  MdocCreateOptions,
+  MdocDeviceResponseOpenId4VpOptions,
+  MdocDeviceResponseVerifyOptions,
+  MdocVerifyOptions,
+} from './MdocOptions'
+import type { Query, QueryOptions } from '../../storage/StorageService'
+
+import { injectable } from 'tsyringe'
+
+import { AgentContext } from '../../agent'
+
+import { Mdoc } from './Mdoc'
+import { MdocDeviceResponse } from './MdocDeviceResponse'
+import { MdocRecord, MdocRepository } from './repository'
+
+/**
+ * @internal
+ */
+@injectable()
+export class MdocService {
+  private MdocRepository: MdocRepository
+
+  public constructor(mdocRepository: MdocRepository) {
+    this.MdocRepository = mdocRepository
+  }
+
+  public mdocFromBase64Url(hexEncodedMdoc: string) {
+    return Mdoc.fromBase64Url(hexEncodedMdoc)
+  }
+
+  public createMdoc(agentContext: AgentContext, options: MdocCreateOptions) {
+    return Mdoc.create(agentContext, options)
+  }
+
+  public async verifyMdoc(agentContext: AgentContext, mdoc: Mdoc, options: MdocVerifyOptions) {
+    return await mdoc.verify(agentContext, options)
+  }
+
+  public async createDeviceResponse(agentContext: AgentContext, options: MdocDeviceResponseOpenId4VpOptions) {
+    return MdocDeviceResponse.openId4Vp(agentContext, options)
```

Comment:
Is device response an openid4vp specific object, if we're using BLE don't we use device response? If so I don't think the underlying API should be called `openId4Vp`?

Reply:
Or should this method be called `createOpenId4VpDeviceResponse`?

Reply:
renamed to createOpenId4VpDeviceResponse